pkg/network/event_common.go

```diff
@@ -414,22 +414,25 @@ func printAddress(address util.Address, names []dns.Hostname) string {
 // This function generates all relevant combinations of connection keys: [(source, dest), (dest, source), (NAT'd source, NAT'd dest), (NAT'd dest, NAT'd source)].
 // This is necessary to handle all possible scenarios for connections originating from the USM module (i.e., whether they are NAT'd or not, and whether they use TLS).
 func ConnectionKeysFromConnectionStats(connectionStats ConnectionStats) []types.ConnectionKey {
-
+	hasTranslation := connectionStats.IPTranslation != nil
+	factor := 1
+	if hasTranslation {
+		factor = 2
+	}
+	connectionKeys := make([]types.ConnectionKey, 2*factor)
```

Comment:
seems like you use the factor only at this part, so can it be set to be 2 and 4 in order to avoid the 2*factor in the call of the function?